error in annotation reply

### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [x] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.16

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

enable the annotated reply, an error occurs when the answer hits the annotated content. It is normal in 0.6.15, but an error occurs in the latest version 0.6.16

When the question does not hit the annotation, everything is OK

<img width="997" alt="image" src="https://github.com/user-attachments/assets/2d016f32-bd55-4de0-a70d-9cbbd74cd6a3">


### ✔️ Expected Behavior

reply  the annotated content

### ❌ Actual Behavior

nothing relpy 

<img width="1143" alt="image" src="https://github.com/user-attachments/assets/39129f67-2fb8-4254-80e4-995dbad84348">


```
[2024-08-07 01:56:33 +0000] [46] [ERROR] Error handling request /api/chat-messages
Traceback (most recent call last):
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/base_async.py", line 55, in handle
    self.handle_request(listener_name, req, client, addr)
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/ggevent.py", line 128, in handle_request
    super().handle_request(listener_name, req, sock, addr)
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/base_async.py", line 115, in handle_request
    for item in respiter:
  File "/app/api/.venv/lib/python3.10/site-packages/werkzeug/wsgi.py", line 256, in __next__
    return self._next()
  File "/app/api/.venv/lib/python3.10/site-packages/werkzeug/wrappers/response.py", line 32, in _iter_encoded
    for item in iterable:
  File "/app/api/.venv/lib/python3.10/site-packages/flask/helpers.py", line 113, in generator
    yield from gen
  File "/app/api/libs/helper.py", line 168, in generate
    yield from response
  File "/app/api/core/app/features/rate_limiting/rate_limit.py", line 110, in __next__
    return next(self.generator)
  File "/app/api/core/app/apps/base_app_generate_response_converter.py", line 37, in _generate_simple_response
    for chunk in cls.convert_stream_simple_response(response):
  File "/app/api/core/app/apps/advanced_chat/generate_response_converter.py", line 94, in convert_stream_simple_response
    for chunk in stream_response:
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 179, in _to_stream_response
    for stream_response in generator:
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 206, in _wrapper_process_stream_response
    for response in self._process_stream_response(publisher=publisher, trace_manager=trace_manager):
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 247, in _process_stream_response
    if hasattr(message.event, 'metadata') and message.event.metadata.get('is_answer_previous_node', False) and publisher:
AttributeError: 'NoneType' object has no attribute 'get'
```

--- a/api/core/app/apps/advanced_chat/generate_task_pipeline.py
+++ b/api/core/app/apps/advanced_chat/generate_task_pipeline.py
@@ -244,7 +244,7 @@ class AdvancedChatAppGenerateTaskPipeline(BasedGenerateTaskPipeline, WorkflowCyc
         :return:
         """
         for message in self._queue_manager.listen():
-            if hasattr(message.event, 'metadata') and message.event.metadata.get('is_answer_previous_node', False) and publisher:
+            if message.event and hasattr(message.event, 'metadata') and message.event.metadata.get('is_answer_previous_node', False) and publisher:
                 publisher.publish(message=message)
             elif (hasattr(message.event, 'execution_metadata')
                   and message.event.execution_metadata

--- a/api/core/rag/datasource/vdb/analyticdb/analyticdb_vector.py
+++ b/api/core/rag/datasource/vdb/analyticdb/analyticdb_vector.py
@@ -285,9 +285,11 @@ class AnalyticdbVector(BaseVector):
         documents = []
         for match in response.body.matches.match:
             if match.score > score_threshold:
+                metadata = json.loads(match.metadata.get("metadata_"))
                 doc = Document(
                     page_content=match.metadata.get("page_content"),
-                    metadata=json.loads(match.metadata.get("metadata_")),
+                    vector=match.metadata.get("vector"),
+                    metadata=metadata,
                 )
                 documents.append(doc)
         return documents

--- a/api/core/rag/datasource/vdb/myscale/myscale_vector.py
+++ b/api/core/rag/datasource/vdb/myscale/myscale_vector.py
@@ -126,13 +126,14 @@ class MyScaleVector(BaseVector):
         where_str = f"WHERE dist < {1 - score_threshold}" if \
             self._metric.upper() == "COSINE" and order == SortOrder.ASC and score_threshold > 0.0 else ""
         sql = f"""
-            SELECT text, metadata, {dist} as dist FROM {self._config.database}.{self._collection_name}
+            SELECT text, vector, metadata, {dist} as dist FROM {self._config.database}.{self._collection_name}
             {where_str} ORDER BY dist {order.value} LIMIT {top_k}
         """
         try:
             return [
                 Document(
                     page_content=r["text"],
+                    vector=r['vector'],
                     metadata=r["metadata"],
                 )
                 for r in self._client.query(sql).named_results()

--- a/api/core/rag/datasource/vdb/opensearch/opensearch_vector.py
+++ b/api/core/rag/datasource/vdb/opensearch/opensearch_vector.py
@@ -192,7 +192,9 @@ class OpenSearchVector(BaseVector):
         docs = []
         for hit in response['hits']['hits']:
             metadata = hit['_source'].get(Field.METADATA_KEY.value)
-            doc = Document(page_content=hit['_source'].get(Field.CONTENT_KEY.value), metadata=metadata)
+            vector = hit['_source'].get(Field.VECTOR.value)
+            page_content = hit['_source'].get(Field.CONTENT_KEY.value)
+            doc = Document(page_content=page_content, vector=vector, metadata=metadata)
             docs.append(doc)
 
         return docs

--- a/api/core/rag/datasource/vdb/oracle/oraclevector.py
+++ b/api/core/rag/datasource/vdb/oracle/oraclevector.py
@@ -234,16 +234,16 @@ class OracleVector(BaseVector):
                         entities.append(token)
             with self._get_cursor() as cur:
                 cur.execute(
-                    f"select meta, text FROM {self.table_name} WHERE CONTAINS(text, :1, 1) > 0 order by score(1) desc fetch first {top_k} rows only",
+                    f"select meta, text, embedding FROM {self.table_name} WHERE CONTAINS(text, :1, 1) > 0 order by score(1) desc fetch first {top_k} rows only",
                     [" ACCUM ".join(entities)]
                 )
                 docs = []
                 for record in cur:
-                    metadata, text = record
-                    docs.append(Document(page_content=text, metadata=metadata))
+                    metadata, text, embedding = record
+                    docs.append(Document(page_content=text, vector=embedding, metadata=metadata))
             return docs
         else:
-            return [Document(page_content="", metadata="")]
+            return [Document(page_content="", metadata={})]
         return []
 
     def delete(self) -> None:

--- a/api/core/rag/datasource/vdb/qdrant/qdrant_vector.py
+++ b/api/core/rag/datasource/vdb/qdrant/qdrant_vector.py
@@ -399,7 +399,6 @@ class QdrantVector(BaseVector):
                 document = self._document_from_scored_point(
                     result, Field.CONTENT_KEY.value, Field.METADATA_KEY.value
                 )
-                document.metadata['vector'] = result.vector
                 documents.append(document)
 
         return documents
@@ -418,6 +417,7 @@ class QdrantVector(BaseVector):
     ) -> Document:
         return Document(
             page_content=scored_point.payload.get(content_payload_key),
+            vector=scored_point.vector,
             metadata=scored_point.payload.get(metadata_payload_key) or {},
         )
 

--- a/api/core/rag/datasource/vdb/weaviate/weaviate_vector.py
+++ b/api/core/rag/datasource/vdb/weaviate/weaviate_vector.py
@@ -239,8 +239,7 @@ class WeaviateVector(BaseVector):
         query_obj = self._client.query.get(collection_name, properties)
         if kwargs.get("where_filter"):
             query_obj = query_obj.with_where(kwargs.get("where_filter"))
-        if kwargs.get("additional"):
-            query_obj = query_obj.with_additional(kwargs.get("additional"))
+        query_obj = query_obj.with_additional(["vector"])
         properties = ['text']
         result = query_obj.with_bm25(query=query, properties=properties).with_limit(kwargs.get('top_k', 2)).do()
         if "errors" in result:
@@ -248,7 +247,8 @@ class WeaviateVector(BaseVector):
         docs = []
         for res in result["data"]["Get"][collection_name]:
             text = res.pop(Field.TEXT_KEY.value)
-            docs.append(Document(page_content=text, metadata=res))
+            additional = res.pop('_additional')
+            docs.append(Document(page_content=text, vector=additional['vector'], metadata=res))
         return docs
 
     def _default_schema(self, index_name: str) -> dict:

--- a/api/core/rag/models/document.py
+++ b/api/core/rag/models/document.py
@@ -10,6 +10,8 @@ class Document(BaseModel):
 
     page_content: str
 
+    vector: Optional[list[float]] = None
+
     """Arbitrary metadata about the page content (e.g., source, relationships to other
         documents, etc.).
     """

--- a/api/core/tools/tool/dataset_retriever/dataset_retriever_tool.py
+++ b/api/core/tools/tool/dataset_retriever/dataset_retriever_tool.py
@@ -78,7 +78,8 @@ class DatasetRetrieverTool(DatasetRetrieverBaseTool):
                                                       top_k=self.top_k,
                                                       score_threshold=retrieval_model.get('score_threshold', .0)
                                                       if retrieval_model['score_threshold_enabled'] else None,
-                                                      reranking_model=retrieval_model.get('reranking_model', None),
+                                                      reranking_model=retrieval_model.get('reranking_model', None)
+                                                      if retrieval_model['reranking_enable'] else None,
                                                       reranking_mode=retrieval_model.get('reranking_mode')
                                                       if retrieval_model.get('reranking_mode') else 'reranking_model',
                                                       weights=retrieval_model.get('weights', None),

--- a/api/services/hit_testing_service.py
+++ b/api/services/hit_testing_service.py
@@ -44,7 +44,8 @@ class HitTestingService:
                                                   top_k=retrieval_model.get('top_k', 2),
                                                   score_threshold=retrieval_model.get('score_threshold', .0)
                                                   if retrieval_model['score_threshold_enabled'] else None,
-                                                  reranking_model=retrieval_model.get('reranking_model', None),
+                                                  reranking_model=retrieval_model.get('reranking_model', None)
+                                                  if retrieval_model['reranking_enable'] else None,
                                                   reranking_mode=retrieval_model.get('reranking_mode')
                                                   if retrieval_model.get('reranking_mode') else 'reranking_model',
                                                   weights=retrieval_model.get('weights', None),

